pkg/config/defaults.go

```diff
@@ -0,0 +1,210 @@
+					CPU:    "500m",
+					Memory: "1Gb",
+				},
+			},
+		},
+		Service: types.LongRunningJobDefaultsConfig{
+			Priority: 0,
+			Task: types.LongRunningTaskDefaultConfig{
+				Resources: types.ResourcesConfig{
+					CPU:    "500m",
+					Memory: "1Gb",
+				},
+			},
+		},
+	},
+	InputSources: types.InputSourcesConfig{
+		ReadTimeout:   5 * types.Minute,
+		MaxRetryCount: 3,
+	},
+	Engines: types.EngineConfig{
+		Types: types.EngineConfigTypes{
+			Docker: types.Docker{
+				ManifestCache: types.DockerManifestCache{
+					Size:    1000,
+					TTL:     types.Duration(1 * time.Hour),
+					Refresh: types.Duration(1 * time.Hour),
+				}},
+		},
+	},
+	Publishers: types.PublishersConfig{
+		Types: types.PublisherTypes{
+			Local: types.LocalPublisher{
+				Address: "127.0.0.1",
+				Port:    6001,
+			},
+		},
+	},
+	JobAdmissionControl: types.JobAdmissionControl{
+		Locality: models.Anywhere,
+	},
+	Logging: types.Logging{
+		Level:                "info",
+		Mode:                 "default",
+		LogDebugInfoInterval: 30 * types.Second,
+	},
+	UpdateConfig: types.UpdateConfig{
+		Interval: types.Day,
+	},
+	WebUI: types.WebUI{
+		Enabled: false,
+		Listen:  "0.0.0.0:8438",
+	},
+}
+
+var testOverrides = types.Bacalhau{
+	API: types.API{
+		Port: -1,
+		Auth: types.AuthConfig{},
+	},
+	Orchestrator: types.Orchestrator{
+		Port: -1,
+		NodeManager: types.NodeManager{
+			DisconnectTimeout: types.Duration(30 * time.Second),
+		},
+		Scheduler: types.Scheduler{
+			WorkerCount:         3,
+			HousekeepingTimeout: types.Duration(100 * time.Millisecond),
+		},
+		EvaluationBroker: types.EvaluationBroker{
+			VisibilityTimeout: types.Duration(5 * time.Second),
+			MaxRetryCount:     3,
+		},
+	},
+	Compute: types.Compute{
+		Heartbeat: types.Heartbeat{
+			Interval: 5 * types.Second,
+		},
+	},
+	Publishers: types.PublishersConfig{
+		Types: types.PublisherTypes{
+			Local: types.LocalPublisher{
+				Port: -1,
+			},
+		},
+	},
+	Logging: types.Logging{
+		Level: "debug",
+	},
+	UpdateConfig: types.UpdateConfig{
+		Interval: -1,
+	},
+	DisableAnalytics: true,
+}
+
+// NewTestConfig returns a new configuration with the default values for testing.
+func NewTestConfig() (types.Bacalhau, error) {
+	cfg, err := Default.MergeNew(testOverrides)
+	if err != nil {
+		return types.Bacalhau{}, err
+	}
+
+	// Set the data directory to a temporary directory
+	cfg.DataDir = os.TempDir()
+
+	return cfg, nil
+}
+
+func NewTestConfigWithOverrides(overrides types.Bacalhau) (types.Bacalhau, error) {
+	cfg, err := NewTestConfig()
+	if err != nil {
+		return types.Bacalhau{}, err
+	}
+
+	return cfg.MergeNew(overrides)
+}
+
+// defaultDataDir determines the appropriate default directory for storing repository data.
+// Priority order:
+// 1. User's home directory with .bacalhau appended.
+func defaultDataDir() string {
+	if userHome, err := os.UserHomeDir(); err == nil && userHome != "" {
+		if expandedUserHome, err := filepath.Abs(userHome); err == nil {
```

Comment:
nit: I don't think we need to expand the home dir. `os.UserHomeDir` returns an absolute path.

Reply:
In your tilda path PR you were validating if `UserHomeDir()` is abs or not, and I assumed you had a reason for that. Should I remove this?
https://github.com/bacalhau-project/bacalhau/pull/4485/commits/0387640d61af302a01e2b06c6daefbe1768a24a4#diff-5e48b782fb0f0b0ef48ec6cfe35aa63eb7d2390c411a9be112652233ba5ccafe

Reply:
Like this method?:
```go
func AbsPathSilent(path string) string {
	expandedPath, err := homedir.Expand(path)
	if err != nil {
		log.Debug().Msgf("failed to expand path %s: %v", path, err)
		return path
	}
	absPath, err := filepath.Abs(expandedPath)
	if err != nil {
		log.Debug().Msgf("failed to resolve absolute path for %s: %v", path, err)
		return path
	}
	return absPath
}
```
Slightly different, am attempting to expand a path containing a tilda, if the path doesn't contain a tilda the call is a noop, we then ensure its an absolute path - mainly in the event Expand was a noop.

Reply:
No I was referring to this if statement you had in `DefaultDataDir()`
```
if userHome, err := os.UserHomeDir(); err == nil && filepath.IsAbs(userHome) {
```